tools/build/Makefile.rules

```diff
@@ -355,14 +360,29 @@ $(SOL_LIB_SO): $(builtin-objs)
 	$(Q)$(LN) -fs $(notdir $(@).$(VERSION)) $(@)
 
 # generators
+define make-oic-gen
+$(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .json) $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .c) $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), -gen.c) $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .h): $(FLOW_OIC_GEN_SCRIPT)
+	$(Q)echo "     "GEN"   "$$@
+	$(Q)$(PYTHON) $(FLOW_OIC_GEN_SCRIPT) \
+		--schema-dir=$(1) \
+		--node-type-json=$(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .json) \
+		--node-type-impl=$(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .c) \
+		--node-type-gen-c=$(addprefix $(notdir $(abspath $(1))), -gen.c) \
+		--node-type-gen-h=$(addprefix sol-flow/, $(addprefix $(notdir $(abspath $(1))), .h)) >/dev/null
+	$(eval cleanup-files += $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .json) $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .c))
+endef
+$(foreach gen,$(sort $(oic-gens)),$(eval $(call make-oic-gen,$(gen))))
+
 define make-gen
 $($(1)-hdr) $($(1)-src) $($(1)-desc): $(1) $(NODE_TYPE_GEN_SCRIPT)
 	$(Q)echo "     "GEN"   "$$@
 	$(Q)$(MKDIR) -p $(dir $($(1)-hdr))
 	$(Q)$(MKDIR) -p $(dir $($(1)-src))
 	$(Q)$(MKDIR) -p $(dir $($(1)-desc))
-	$(Q)$(PYTHON) $(NODE_TYPE_GEN_SCRIPT) --prefix=sol-flow-node-type --namespace=$($(1)-namespace) \
-		--genspec-schema=$(NODE_TYPE_SCHEMA) $(1) $($(1)-hdr) $($(1)-src) $($(1)-desc)
+	$(Q)$(PYTHON) $(NODE_TYPE_GEN_SCRIPT) \
+		--prefix=sol-flow-node-type --namespace=$($(1)-namespace) \
+		--genspec-schema=$(NODE_TYPE_SCHEMA) $(1) $($(1)-hdr) \
+		$($(1)-src) $($(1)-desc)
```

Comment:
More than ever, when dealing with our Makefiles, do NOT mix style changes with functional ones.